jax/numpy/lax_numpy.py

```diff
@@ -3140,6 +3140,9 @@ def _index_to_gather(x_shape, idx):
     # Handle basic int indexes.
     if (isinstance(abstract_i, ConcreteArray) or
         isinstance(abstract_i, ShapedArray)) and _int(abstract_i):
+      if x_shape[x_axis] == 0:
```

Comment:
I would have liked to put this further up, but it seems that x[None], x[:] work even for empty axes.